### ISSUE TITLE
Support for threads

### ISSUE THREAD

**Issue by twoscomplement** [timhutton/twitter-archive-parser#23]
It would be nice to be able to extract or format tweet threads in some fashion that makes them easy to identify - right now, all the tweets appear in the output, but not with any explicit indication of connection between the related replies-to-self.

At least one complication is that threads are not necessarily a single linear sequence.

**Comment by timhutton** [comment]
The JSON has the id of the tweet being replied to, if any. But the archive only contains tweets sent by the user, so linking them together might be impossible, unless we start trying to pull content through the API.

**Comment by twoscomplement** [comment]
To clarify, I was (initially, at least :slightly_smiling_face:) looking for an easy way to group reply-to-self style tweet threads - which would be useful to me, independent of having full threads of discussion.

A concrete example being the ~30 tweets starting [here](https://twitter.com/twoscomplement/status/1580380783767756801).

(including twitter.com links to other accounts' tweets being replied to, in output.md, could be useful, too)

**Comment by timhutton** [comment]
Do you have any thoughts on how the threads should be represented in the output?

**Comment by Sjors** [comment]
It would be nice to at least reconstruct the simplest possible threads. One approach could be to group them in the same `----` block. Tweets outside the thread that happened while the thread was developing (which is very common if you occasionally append something to an old thread) would then go to the next `----` block.

**Comment by Sjors** [comment]
As for threads with other people in it, I think there's two variations:

1. My tweet(s), followed by replies (which include my own). Perhaps they're best rendered with block quotes:

```
----

My brilliant hot take

(Originally on Twitter: …)

My own followup tweet

(Originally on Twitter: …)

Reply by @ somebody:

> Wow that's brilliant!

(Originally on Twitter: …)

Thanks!

(Originally on Twitter: …)

```

2. My replies to other tweets

Those should probably go in a separate file, e.g. `interactions_2007_10.md`

```
Tweet by @ somebody
> How much is 2 x 2?

(Originally on Twitter: …)

Reply by @ someone_else

> 5

(Originally on Twitter: …)

4

(Originally on Twitter: …)
```

**Comment by hyc** [comment]
Would definitely be cool to be able to use the API to fetch other peoples' tweets that you're replying to, so you have the complete conversation.

**Comment by timhutton** [comment]
@hyc Yes, perhaps. Please feel free to make a new feature request for it.
